mathesar_ui/src/component-library/number-input/number-formatter/NumberFormatter.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  InputFormatter,
+  ParseResult,
+} from '@mathesar-component-library-dir/common/utils/InputFormatter';
+import { factoryToNormalize } from './cleaners';
+import { makeFormatter } from './formatter';
+import type { DerivedOptions, Options } from './options';
+import { defaultOptions, getDerivedOptions } from './options';
+
+function getParsedValue(normalizedInput: string): number | undefined {
+  // Why use `parseFloat` instead of `parseInt` in cases where we know we should
+  // have an Int? Because we already strip out the decimal separator when
+  // normalizing the input for an expected integer, and all numbers are floats
+  // in javascript anyway.
+  const value = parseFloat(normalizedInput);
+  if (Number.isNaN(value)) {
+    return undefined;
+  }
+  return value;
+}
+
+export default class NumberFormatter implements InputFormatter<number> {
```

Comment:
@seancolsen I think the number formatter should always return a JSON number compliant string instead of a number, when parsed.

In this scenario, assume if the user enters the following:
```
parse("123456.232423124353445"): 123456.23242312435
parse("123456232423124353445): 123456232423124350000
```

Notice that we lose precision in double precision and big integers. Also, when performing `parseFloat` or `parseInt` there are several other cases where we would get lossy values.

I think it's best if parsing only produces a JSON number compliant string by cleaning the string up, stripping down separators and only using `.` as the decimal separator, and we use that wherever user input is required.

Reply:
Thanks for pointing this out, @pavish. This limitation has been in the back of my mind, but I haven't put enough thought yet into dealing with high-precision numbers. I'll think on this more and follow up, either by adopting your suggestion or by proposing something else.

Reply:
@pavish Although this PR still needs more work, the changes I just pushed should address your concern. Now there's a `StringifiedNumberFormatter` and a `StringifiedNumberInput`.

per your agreement here https://github.com/centerofci/mathesar-wiki/pull/39#discussion_r828113600 I'm guessing this new approach will be cool with you. If so, can you resolve this thread?